sources/mapping_functions/bcknd/device/cuda/RAMP_mapping_kernel.h

```diff
@@ -0,0 +1,97 @@
+/*
+ Copyright (c) 2021-2023, The Neko Authors
+ All rights reserved.
+
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions
+ are met:
+
+   * Redistributions of source code must retain the above copyright
+     notice, this list of conditions and the following disclaimer.
+
+   * Redistributions in binary form must reproduce the above
+     copyright notice, this list of conditions and the following
+     disclaimer in the documentation and/or other materials provided
+     with the distribution.
+
+   * Neither the name of the authors nor the names of its
+     contributors may be used to endorse or promote products derived
+     from this software without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#ifndef __NEKO_CUDA_RAMP_MAPPING_KERNELS__
+#define __NEKO_CUDA_RAMP_MAPPING_KERNELS__
+
+/**
+ * Device kernel for RAMP (convex down) mapping
+ */
+template <typename T>
+__global__ void convex_down_RAMP_mapping_apply_kernel(
+    const T f_min, const T f_max, const T q, T* __restrict__ X_out_d,
+    T* __restrict__ X_in_d, const int n) {
+
+    const int idx = blockIdx.x * blockDim.x + threadIdx.x;
+    const int str = blockDim.x * gridDim.x;
+
+    for (int i = idx; i < n; i += str) { X_out_d[i] = f_min + (f_max - f_min)
+    * X_in_d[i] / (1.0 + q * (1.0 + X_in_d[i])); }
```

Comment:
Why is this one line
```suggestion
    for (int i = idx; i < n; i += str) {
        X_out_d[i] = f_min 
            + (f_max - f_min) * X_in_d[i] / (1.0 + q * (1.0 + X_in_d[i]));
    }
```